### DISCUSSION TITLE
so many attributes....

### DISCUSSION THREAD

**Discussion by dustymc** [ArctosDB/arctos#6179] [General]
# Options

As of May 2026 we have not implemented anything consistent. Options seem to be:

1. Add lots (maybe millions, Arctos is very diverse) or individual Attributes to Arctos. This seems to have little support and clearly has major usability implications (eg a student will never find the correct thing in amongst 10 - or 1000 - very similar things.)
2. Store these data in some external "AttributeBank." Possibly the theoretically best option, Arctos has an API so if good identifiers were used it would be easy to tie everything back together again, but no such thing seems to exist.
3. Store attributes in some (new) JSON attribute. This would provide nearly unlimited "local" storage, have little usability cost, and could be implemented immediately. Users would need to ensure valid JSON was supplied in some way (eg by providing some schema, UI, etc., for entry), which might be an insurmountable barrier.
4. Use existing attribute https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#unformatted_measurements - much like Option 3, but without any checks; easier in, possibly much more difficult out.

------------------

# Original
Between some recent projects, data like https://github.com/ArctosDB/arctos/issues/6175, similar data for marine mammals, etc., it seems that we're destined to have hundreds (thousands?!) of attributes. 

https://github.com/ArctosDB/arctos/issues/3452 may (eventually) make it easier for eg a rodent-centric mammal collection to avoid being overwhelmed with cetacean-specific attributes.

https://github.com/ArctosDB/arctos/issues/6111 is the start of a conversation regarding dealing with attributes in tabular form.

@Jegelewicz and I can find no realistic alternatives to just creating attributes as they're requested. This Issue is an acknowledgement that we see the slippery slope and are not sure what's at the bottom of the cliff.

Any alternative viewpoints or approaches, concerns, thoughts, illumination, etc., greatly appreciated!

**Comment by Jegelewicz** [comment]
Just adding that once the RANGES stuff is added and if the hundreds of measurements in #4533 plus this antler stuff are added - the list of mammal attributes will be pretty BIG. Asking people to scroll through it might be frustrating or intimidating OR maybe it is fine? Just start typing the attribute you want and get there?

**Comment by campmlc** [comment]
Just typing the start of a value and having the dropdown go there would be a big help. . . possible?

**Comment by dustymc** [comment]
Possibility: store stuff as JSON - would allow any level of complexity, take only one columns. Would need UI/schema/??? development.

**Comment by dustymc** [comment]
20241126 data dump:

```
select
    ctattribute_type.attribute_type,
    count(attributes.collection_object_id) usecount
from
    ctattribute_type
    left outer join attributes on ctattribute_type.attribute_type=attributes.attribute_type
group by ctattribute_type.attribute_type order by count(*);


        attribute_type        | usecount 
------------------------------+----------
 format                       |        0
 tail girth                   |        0
 antler diameter              |        1
 antler spread                |        1
 antler circumference         |        1
 antler points                |        2
 luster                       |        3
 mortality cause              |        3
 year class                   |        4
 tooth width                  |        7
 chromatography compound      |        8
 middle toe length            |       10
 bursa width                  |       17
 tail base width              |       17
 spleen length                |       19
 neck width                   |       20
 isotope value                |       30
 body width                   |       45
 brood patch                  |       57
 hallux length                |       60
 nest phenology               |       62
 eggshell thickness           |       73
 language                     |       73
 clutch size of nest parasite |       80
 gape width                   |       85
 exhibit caption              |       91
 hind limb length             |       99
 curved carapace length       |      125
 radiometric date             |      137
 straight carapace length     |      150
 electrical conductivity      |      179
 carapace length              |      181
 keywords                     |      267
 plastron length              |      275
 user contributed comments    |      319
 tested for presence          |      334
 diploid number               |      475
 tooth length                 |      489
 breadth                      |      490
 crop contents                |      535
 calcar length                |      559
 thumb length with claw       |      622
 head width                   |      671
 hind foot without claw       |      715
 ovum                         |      723
 NAGPRA category              |      726
 crystal habit                |      762
 historical                   |      822
 bursa length                 |      832
 object title                 |      941
 behavior                     |     1128
 bill width                   |     1130
 bill depth                   |     1165
 width                        |     1307
 parts examined               |     1434
 abundance                    |     1624
 frame rate                   |     1628
 axillary girth               |     1635
 duration                     |     1653
 portfolio or series          |     1672
 curvilinear length           |     1754
 bill length                  |     1863
 body temperature             |     2279
 tail condition               |     2312
 verbatim host age            |     2672
 image confirmed              |     2704
 culmen length                |     3027
 gonad width                  |     3067
 gonad length                 |     3307
 gonad weight                 |     3834
 ear from crown               |     3879
 nest description             |     4033
 copyright notice             |     4326
 framed                       |     4439
 copyright status             |     4570
 caste                        |     4669
 inscriptions and marks       |     5314
 tarsus length                |     5407
 wing span                    |     5480
 fork length                  |     5756
 crown-rump length            |     5898
 experimental                 |     6414
 credit line                  |     6415
 maximum total length         |     6942
 tragus length                |     7772
 dimensions                   |     8780
 forearm length               |    11783
 head length                  |    12521
 Indigenous term              |    13802
 wing chord                   |    15494
 verbatim attribute           |    15755
 culture of use               |    15980
 associated species           |    16696
 incubation stage             |    16983
 subject matter               |    17872
 height                       |    18225
 body condition               |    21521
 bursa                        |    21790
 clutch size                  |    22865
 value                        |    24200
 numeric age                  |    26489
 stomach contents             |    27393
 minimum total length         |    29498
 verbatim preservation date   |    36369
 standard length              |    36539
 skull ossification           |    37110
 snout-vent length            |    41709
 detected                     |    42970
 not examined for             |    45086
 verbatim host sex            |    45736
 colors                       |    50640
 molt condition               |    70032
 archaeological feature       |    74354
 fat deposition               |    79561
 not detected                 |    79949
 maximum standard length      |    89386
 minimum standard length      |    94138
 number of labels             |   113071
 examined for                 |   120853
 provenience east             |   123109
 quadrant                     |   129387
 location in host             |   131443
 unformatted measurements     |   131693
 provenience north            |   134344
 pit depth                    |   158540
 verbatim host ID             |   163633
 culture of origin            |   205684
 provenience                  |   212343
 processing history           |   256633
 ear from notch               |   303725
 age                          |   321539
 stratigraphic position       |   356300
 reproductive data            |   378706
 hind foot with claw          |   381545
 square                       |   400556
 tail length                  |   417233
 total length                 |   438351
 description                  |   478646
 individual count             |   564971
 weight                       |   607902
 materials                    |   619383
 life stage                   |  1150907
 verbatim agent               |  1514502
 sex                          |  2298508

```

**Comment by dustymc** [comment]
See also https://github.com/ArctosDB/code-table-work/issues/79 - I am still having difficulty determining a direction.

**Comment by dustymc** [comment]
See also https://github.com/ArctosDB/arctos/issues/8981 - does having 3 (or 4 - nevermind, 5, I found fork length buried in the clutter!) "length, plus some method/thing in type" Attributes for fish make sense; do the benefits of that specificity outweigh the costs of having 4 (more!) things for users to get lost in? 

Does the answer to that say anything about...

* https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#straight_carapace_length
* https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#snout_vent_length
* https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#curvilinear_length
* https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#fork_length
* https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#curved_carapace_length
* https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#carapace_length


... and whatever else I missed.

Ideally this discussion would produce something prescriptive: something we could point at and act immediately when the next 'length measured thusly' request comes in.


In case anyone cares:  I still don't have much opinion, we should do whatever best balances supporting research and usability. The research aspect is moot if the students are choosing type arbitrarily, and the data suggests they occasionally do as the number of THINGs increases. Researchers can and should be expected to dig though method, but probably not very deeply. I'm not at all sure where that point is. My weak inclination is (usually, I think) towards fewer types and more data in method, but I'd not be too surprised to find that's not the best direction. I would probably support any sort of prescriptive policy; being stuck in limbo often seems worse than any of the alternatives.

https://handbook.arctosdb.org/best_practices/committee.html please....

**Comment by zygoballus** [comment]
If we end up moving our ethnographic, archaeological, and mineral collections into Arctos, I will have a whole slew of new attributes to request, both for objects and parts. If you want a rough draft of that list, let me know.

**Comment by dustymc** [comment]
>whole slew of new attributes

This is one reason I tend to prefer fewer THINGS (and why i occasionally think that nobody should do anything else until all authorities are cleaned up and fully documented!!): incoming data often includes just terms, these often appear to be different ways of saying familiar things or duplicates of each other, or etc., etc., etc., and there's often a time constraint; resolving all of that sometimes/often/please-dont-be-always finds some way to add to the clutter. (BUT, maybe a policy of 'add anything that has a functional definition and is distinct from any existing concept' would be as good or better???) Any sort of policy that smooths that road would be most welcome. 

>rough draft

Sure, maybe it'll help guide some discussion, thanks!

**Comment by dustymc** [comment]
From @zygoballus in https://github.com/ArctosDB/arctos/issues/8981#issuecomment-3228741714, bringing it over here as a useful general consideration:

>Total length and standard length are also different for many types of arthropods. That said, it would be nice to have just a "length" attribute to go with the "width" and "height" attributes comprising standard dimensions (especially for non-biological objects). I think querying on any length attribute is going to be a very rare use case, so I doubt it matters much if there are several versions.

Query is the only payback for dealing with the complexity; if structured and conceptualized data aren't useful for query then they're probably better stored as https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#unformatted_measurements, which is super-simple for "us" (copy/paste/transcribe from your source data!) and as useful as anything else to a user who's already found the record. 'As simple as possible, as complex as necessary' tends to be a pretty good guide; perhaps we could incorporate some sort of 'necessary for {reasons}' criteria into any incoming requests. (Highly hypothetical flip side: we ain't never gonna know that goldfish eyelash length is in fact useful for research if we don't make it available to researchers.)
